framerail/src/routes/[x+2d]/login/+server.ts

```diff
@@ -0,0 +1,31 @@
+import { authLogin } from "$lib/server/auth/login"
+
+export async function POST(event) {
+  let data = await event.request.formData()
+
+  let userAgent = event.request.headers.get("User-Agent")
+  let ipAddress = event.getClientAddress()
+
+  let nameOrEmail = data.get("name-or-email")?.toString()
+  let password = data.get("password")?.toString()
+
+  try {
+    let res = await authLogin(nameOrEmail, password, ipAddress, userAgent)
+
+    if (res.session_token) {
+      event.cookies.set("wikijump_token", res.session_token, {
+        path: "/"
```

Comment:
We should set some other flags here:
* `httpOnly: true` - client-side JS has no need to read the cookie itself, just have it sent with any requests. greatly mitigates the risk of XSS vulnerabilities
* `sameSite: 'lax'` - based on [this description](https://owasp.org/www-community/SameSite), `Lax` seems like the proper choice
* `secure: true` - only sent via HTTPS. Wikijump will only support HTTPS so this won't cause issues, and resists ability to sniff and steal session cookies
* `maxAge: (..)` - I guess I should modify the RPC call for session creation to return the expiration date so it can be passed in here. for now add a TODO?